x-pack/legacy/plugins/siem/public/pages/case/components/add_comment/index.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useCallback } from 'react';
+import { EuiButton, EuiLoadingSpinner } from '@elastic/eui';
+import styled from 'styled-components';
+import { Form, useForm } from '../../../shared_imports';
+import { NewComment } from '../../../../containers/case/types';
+import { usePostComment } from '../../../../containers/case/use_post_comment';
+import { schema } from './schema';
+import * as i18n from '../../translations';
+import { MarkdownEditor } from '../markdown_editor';
+
+const MySpinner = styled(EuiLoadingSpinner)`
+  position: absolute;
+  top: 50%;
+  left: 50%;
+`;
+
+export const AddComment = React.memo<{
+  caseId: string;
+}>(({ caseId }) => {
+  const [{ data, isLoading, newComment }, setFormData] = usePostComment(caseId);
+  const { form } = useForm({
+    defaultValue: data,
+    options: { stripEmptyFields: false },
+    schema,
+  });
+
+  const onSubmit = useCallback(async () => {
+    // @XavierM this is weird, why is it isValid when returning an {} w my schema??
```

Comment:
@XavierM this is the form hook issue i wanted to talk about

Reply:
steps to reproduce: fill out the comment, switch tab to "Preview" and submit. put a `log` on line 35 and see that `newData` comes back as `{}` and `isValid === true`. If you do this on the create page, (enter content and flip the markdown component to preview) you'll see that the markdown field, `description`, is excluded from `newData` while other keys are there.

Reply:
this happens because the `CommonUseField` is removed from the page when the Preview tab is active. idk why it would come up as `isValid`. we can keep the `CommonUseField` in the Preview tab also and hide it? idk what is a clean way to fix